x-pack/plugins/security_solution/server/lib/detection_engine/rule_response_actions/utils.ts

```diff
@@ -0,0 +1,118 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get } from 'lodash';
+import type { AlertAgent, AlertWithAgent, AlertsAction } from './types';
+import type { ProcessesParams } from '../../../../common/api/detection_engine';
+
+export const getProcessAlerts = (alerts: AlertWithAgent[], config: ProcessesParams['config']) => {
+  if (!config) {
+    return {};
+  }
+  const { overwrite, field } = config;
+
+  return alerts.reduce((acc: Record<string, Record<string, AlertsAction>>, alert) => {
+    const valueFromAlert: number = overwrite ? alert.process?.pid : get(alert, field);
+
+    if (valueFromAlert) {
+      const isEntityId = !overwrite && field.includes('entity_id');
+      const paramKey = isEntityId ? 'entity_id' : 'pid';
+      const { _id, agent } = alert;
+      const { id: agentId, name } = agent as AlertAgent;
+      const hostName = alert.host?.name;
+
+      const currentAgent = acc[agentId];
+      const currentValue = currentAgent?.[valueFromAlert];
+
+      return {
+        ...acc,
+        [agentId]: {
+          ...(currentAgent || {}),
+          [valueFromAlert]: {
+            ...(currentValue || {}),
+            alert_ids: [...(currentValue?.alert_ids || []), _id],
+            parameters: { [paramKey]: valueFromAlert },
+            endpoint_ids: [agentId],
+            hosts: {
+              ...currentValue?.hosts,
+              [agentId]: { name: name || hostName, id: agentId },
+            },
+          },
+        },
+      };
+    }
+    return acc;
+  }, {});
+};
+
+export const getErrorProcessAlerts = (
+  alerts: AlertWithAgent[],
+  config: ProcessesParams['config']
+) => {
+  if (!config) {
+    return {};
+  }
+  const { overwrite, field } = config;
+
+  return alerts.reduce((acc: Record<string, Record<string, AlertsAction>>, alert) => {
+    const valueFromAlert: number = overwrite ? alert.process?.pid : get(alert, field);
+
+    if (!valueFromAlert) {
+      const isEntityId = !overwrite && field.includes('entity_id');
+      const paramKey = isEntityId ? 'entity_id' : 'pid';
+      const { _id, agent } = alert;
+      const { id: agentId, name } = agent as AlertAgent;
+      const hostName = alert.host?.name;
+
+      const errorField = overwrite ? 'process.pid' : field;
+      const currentAgent = acc[agentId];
+      const currentValue = currentAgent?.[errorField];
+
+      return {
+        ...acc,
+        [agentId]: {
+          ...(currentAgent || {}),
+          [errorField]: {
+            ...(currentValue || {}),
+            alert_ids: [...(currentValue?.alert_ids || []), _id],
+            parameters: { [paramKey]: `${field || 'process.pid'} not found` },
+            endpoint_ids: [agentId],
+            hosts: {
+              ...currentValue?.hosts,
+              [agentId]: { name: name || hostName, id: agentId },
+            },
+            error: errorField,
```

Comment:
So error is always being set?

Reply:
Yes, in this case - the function is calculating only 'failing' actions - because of missing value at alert's ecs field. 
What do you think about it?

Reply:
ahhh, ok. so its only failures with means error is actually meant to be set. I'm good. Thanks.